src/modules/85-iacm/components/IACMSideNav/__tests__/IACMSideNav.test.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React from 'react'
+import { render, waitFor, fireEvent } from '@testing-library/react'
+import { TestWrapper } from '@common/utils/testUtils'
+import routes from '@common/RouteDefinitions'
+import { accountPathProps, orgPathProps, pipelineModuleParams, projectPathProps } from '@common/utils/routeUtils'
+import IACMSideNav from '..'
+
+jest.mock('@projects-orgs/components/ProjectSelector/ProjectSelector', () => ({
+  ProjectSelector: function ProjectSelectorComp(props: any) {
+    return (
+      <button
+        onClick={() => props.onSelect({ identifier: 'project', orgIdentifier: 'org' })}
+        id="projectSelectorId"
+      ></button>
+    )
+  }
+}))
+
+describe('Sidenav', () => {
+  test('render', () => {
+    const { container } = render(
+      <TestWrapper
+        path="/account/:accountId/iacm/orgs/:orgIdentifier/projects/:projectIdentifier"
+        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
+      >
+        <IACMSideNav />
+      </TestWrapper>
+    )
+    expect(container).toMatchSnapshot()
+  })
+
+  test('should go to dashboard when project is selected', async () => {
+    const { container, getByTestId } = render(
+      <TestWrapper
+        path={routes.toIACMStacks({
+          ...accountPathProps,
+          ...orgPathProps,
+          ...pipelineModuleParams,
+          ...projectPathProps
+        })}
+        pathParams={{
+          accountId: 'accountId',
+          orgIdentifier: 'default',
+          projectIdentifier: 'iacm',
+          module: 'iacm'
+        }}
+      >
+        <IACMSideNav />
+      </TestWrapper>
+    )
+
+    const projectButtonSel = '#projectSelectorId'
+    const projectButton = await waitFor(() => container.querySelector(projectButtonSel))
+    fireEvent.click(projectButton!)
```

Comment:
`waitFor` is not really intended to be used like this. It should be used to make an assertion, not a selection. If you change the `id` to a `data-testid`, then you can change this to a single line.

```suggestion
    userEvent.click(getByTestId('projectSelectorId'))
```